File: src/main/java/de/saschahlusiak/freebloks/game/AddScoreTask.java
package de.saschahlusiak.freebloks.game;

import android.content.Context;
import android.database.sqlite.SQLiteException;
import android.os.AsyncTask;
import com.crashlytics.android.Crashlytics;
import de.saschahlusiak.freebloks.model.GameMode;
import de.saschahlusiak.freebloks.model.PlayerScore;
import de.saschahlusiak.freebloks.database.HighscoreDB;

public class AddScoreTask extends AsyncTask<PlayerScore,Void,Void> {
	private GameMode game_mode;
	private Context context;

	public AddScoreTask(Context context, GameMode game_mode) {
		this.game_mode = game_mode;
		this.context = context;
	}

	@Override
	protected Void doInBackground(PlayerScore... data) {
		HighscoreDB db = new HighscoreDB(context);
		if (data == null)
			return null;

		try {
			db.open();

			for (int i = 0; i < data.length; i++) if (data[i].isLocal()) {
				int flags = 0;
				if (data[i].isPerfect())
					flags |= HighscoreDB.FLAG_IS_PERFECT;

				db.addHighscore(
					game_mode,
					data[i].getPoints(),
					data[i].getStonesLeft(),
					data[i].getPlayer1(),
					data[i].getPlace(),
					flags);
			}

			db.close();
		}
		catch (SQLiteException e) {
			Crashlytics.logException(e);
			e.printStackTrace();
		}
		return null;
	}
}

File: src/main/java/de/saschahlusiak/freebloks/game/AddScoreTask.kt
package de.saschahlusiak.freebloks.game

import android.content.Context
import android.database.sqlite.SQLiteException
import com.crashlytics.android.Crashlytics
import de.saschahlusiak.freebloks.database.HighscoreDB
import de.saschahlusiak.freebloks.model.GameMode
import de.saschahlusiak.freebloks.model.PlayerScore

class AddScoreTask(private val context: Context, private val gameMode: GameMode, private val scores: Array<PlayerScore>) : Thread() {
    override fun run() {
        val db = HighscoreDB(context)

        try {
            db.open()

            scores
                .filter { it.isLocal }
                .forEach { score ->
                    var flags = 0
                    if (score.isPerfect) flags = flags or HighscoreDB.FLAG_IS_PERFECT

                    db.addHighscore(
                        gameMode,
                        score.totalPoints,
                        score.stonesLeft,
                        score.color1,
                        score.place,
                        flags
                    )

                }

            db.close()
        } catch (e: SQLiteException) {
            Crashlytics.logException(e)
            e.printStackTrace()
        }
    }

}